documentation/guides/migrating-from-v3-to-v4.md

```diff
@@ -0,0 +1,390 @@
+
+**Note:** This migration guide is for v3 but may apply for v2 as well.
+
+- [Testing](#testing)
+- [API Changes](#api-changes)
+  - [AppProvider](#appprovider)
+    - [i18n](#i18n)
+  - [Navigation](#navigation)
+    - [iconBody](#icon-body)
+  - [ChoiceList](#choicelist)
+    - [title](#title)
+  - [Card](#card)
+    - [secondaryFooterAction](#secondaryfooteraction)
+  - [Select](#select)
+    - [groups](#groups)
+  - [Icon](#icon)
+    - [source](source)
+    - [untrusted](#untrusted)
+- [Removed Exports/Components](#removed-exportscomponents)
+  - [WithContext](#withcontext)
+  - [WithRef](#withref)
+  - [Autocomplete.ComboBox.TextField](#autocompletecomboboxtextfield)
+  - [Autocomplete.ComboBox.OptionList](#autocompletecomboboxoptionlist)
+  - [Navigation.UserMenu](#navigationusermenu)
+  - [Modal.Dialog](#modaldialog)
+  - [Tabs.Panel](#tabspanel)
+- [Dependencies](#dependencies)
+- [Typescript](#typescript)
+  - [Config](#config)
+  - [Types](#types)
+
+## Testing
+
+We've migrated to react new context API as well as restructured our entire context structure. To keep up to date using our test provider.
+
+In v3, you could hook into our context types.
+
+**Note:** These examples use enzyme. See [our examples](https://github.com/Shopify/polaris-react/tree/master/examples) for other testing libraries.
+
+```jsx
+// old
+import merge from 'lodash/merge';
+import {shallow, mount} from 'enzyme';
+import {createPolarisContext, polarisContextTypes} from '@shopify/polaris';
+
+function mergeAppProviderOptions(options) {
+  const context = createPolarisContext();
+
+  return merge(
+    {},
+    {
+      context,
+      childContextTypes: polarisContextTypes,
+    },
+    options,
+  );
+}
+
+export function mountWithAppProvider(node, options) {
+  return mount(node, mergeAppProviderOptions(options));
+}
+
+export function shallowWithAppProvider(node, options) {
+  return shallow(node, mergeAppProviderOptions(options)).dive(options);
+}
+```
+
+Moving forward you'll want to wrap your code in our test provider.
+
+```jsx
+// new
+import {mount} from 'enzyme';
+import {PolarisTestProvider} from '@shopify/polaris';
+
+export function mountWithPolaris(node) {
+  return mount<P>(
+    <PolarisTestProvider {...context}>{node}</PolarisTestProvider>,
+  );
+}
+```
+
+## API Changes
+
+### AppProvider
+
+#### i18n
+
+i18n is now a required prop and Polaris includes [many translation](https://github.com/Shopify/polaris-react/blob/master/locales) to support internationalization.
+
+```jsx
+import fr from '@shopify/polaris/locales/fr.json';
+...
+<AppProvider i18n={fr}>
+...
+```
+
+### Navigation
+
+#### iconBody
+
+<!-- ping ben or andre about this <svg><path d='M17 9h-6V3a1 1 0 1 0-2 0v6H3a1 1 0 1 0 0 2h6v6a1 1 0 1 0 2 0v-6h6a1 1 0 1 0 0-2'  fill-rule='evenodd'/></svg> doesnt seem to work -->
```

Comment:
cc @amrocha (in case you haven't been pinged yet)

Reply:
@AndrewMusgrave I'm not sure I understand what the issue here is, can you give an example?

Reply:
I may be using the API wrong but string's dont seem to work for iconBody or icon#source

```jsx
function Playground() {
  const icon =
    "<svg><path d='M17 9h-6V3a1 1 0 1 0-2 0v6H3a1 1 0 1 0 0 2h6v6a1 1 0 1 0 2 0v-6h6a1 1 0 1 0 0-2'  fill-rule='evenodd'/></svg>";
  return (
    <Fragment>
      <Icon source={icon} />
      <Navigation location="/">
        <Navigation.Section
          items={[
            {
              url: '/path/to/place',
              label: 'Products',
              icon,
            },
          ]}
        />
      </Navigation>
    </Fragment>
  );
}
```

Reply:
I think that's just something wrong with the SVG you're using, when I updated it to the following SVG it started working again:
```
<svg viewBox='0 0 20 20' xmlns='http://www.w3.org/2000/svg'><path d='M10.707 17.707l5-5a.999.999 0 1 0-1.414-1.414L11 14.586V3a1 1 0 1 0-2 0v11.586l-3.293-3.293a.999.999 0 1 0-1.414 1.414l5 5a.999.999 0 0 0 1.414 0' /></svg>
```

Reply:
TIL: `xmlns` is required for image/svg+xml files which is how we server string icons

Reply:
ah yeah, it needs be XML compliant